lib/vanagon/component/source.rb

```diff
@@ -76,23 +76,23 @@ def parse_and_rewrite(uri)
         # @return [Vanagon::Component::Source] the correct subtype for the given source
         def source(uri, **options) # rubocop:disable Metrics/AbcSize
           # First we try git
-          if Vanagon::Component::Source::Git.valid_remote?(parse_and_rewrite(uri))
+          if (options[:source_type] == 'git') || (options[:source_type].nil? && Vanagon::Component::Source::Git.valid_remote?(parse_and_rewrite(uri)))
```

Comment:
Just throwing this in there based on our IRL conversation, but the `Git` source type should already support local `file://` and `/path/to/some/directory` URIs; this may not be needed at all if a component's source is already a local Git repo.